script.js

```diff
@@ -1,88 +1,69 @@
 let books = [
-    {
-   titulo: '100 años de soledad',
-   author: 'Gabriel Garcia Marquez'
-    },
-    {
-   titulo: 'El tirano',
-   author: 'Massimo Manfredy'
-    }
-]
-
-const bookList = document.querySelector('.books-display');
-let displ = ''
-books.forEach((el, index) => {    
-    displ += `
-    <div class="oneBook" id="book${index}">
-    <h4> ${el.titulo} </h4>
-    <h4> ${el.author} </h4> 
-    <button class="btn-display" id="${index}">remove</button> 
-    </div>
-    ` 
-
-});
-bookList.innerHTML = displ
-let removIds = [...document.querySelectorAll('.btn-display')].map(item => item.id)
-
-const newB = document.getElementById('bk')
-
-newB.addEventListener('submit', function(event){
+  {
+    titulo: '100 años de soledad',
+    author: 'Gabriel Garcia Marquez',
+  },
+  {
+    titulo: 'El tirano',
+    author: 'Massimo Manfredi',
+  },
+];
+
+function storage(ok) {
+  window.localStorage.setItem('localBooks', JSON.stringify(ok));
+}
 
-    event.preventDefault();
-    const reciTi = document.getElementById("recibe-ti").value
-    const reciAu = document.getElementById("recibe-au").value
+function getBooks() {
+  books = JSON.parse(window.localStorage.getItem('localBooks'));
+}
 
-    if(reciTi !== '' && reciAu !== ''){
-    let newBook = {
-        titulo: reciTi,
-        author: reciAu
-    }
-    books.unshift(newBook)
+getBooks();
 
-    let displ = ''
-    books.forEach((el, index) => {    
+function display() {
+  const bookList = document.querySelector('.books-display');
+  let displ = '';
+  books.forEach((el, index) => {
     displ += `
     <div class="oneBook" id="book${index}">
     <h4> ${el.titulo} </h4>
     <h4> ${el.author} </h4> 
     <button class="btn-display" id="${index}">remove</button> 
     </div>
-    `
-
-    });
-    bookList.innerHTML = displ
-
-    console.log(books)
+    `;
+  });
+  bookList.innerHTML = displ;
 }
-})
-
-/* document.querySelectorAll('.btn-display').forEach( (rem) => {
-    rem.addEventListener('click', (event) => {
-        console.log(rem)
-    })
 
-}) */
+// ---------ADD-------------
+
+const newB = document.getElementById('bk');
+newB.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const reciTi = document.getElementById('recibe-ti').value;
+  const reciAu = document.getElementById('recibe-au').value;
+
+  if (reciTi !== '' && reciAu !== '') {
+    const newBook = {
+      titulo: reciTi,
+      author: reciAu,
+    };
+    books.unshift(newBook);
+    storage(books);
+    display();
+  }
+});
 
-function selectBook(e){
-    
+// -------DELETE----------
+function deleteBook(evento) {
+  books = books.filter((el, index) => evento.target.id !== index.toString());
+  storage(books);
+  display();
```

Comment:
- [ ] It has a function that removes a book from the collection (hint: students can use the [array filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method). As you have the function but due to not working of adding function, it is also not working. kindly check it